synapseclient/table.py

```diff
@@ -829,75 +857,155 @@ class Dataset(SchemaBase):
     :py:classmethod:`synapseclient.client.create_snapshot_version`.
 
     Example::
+
         syn = synapseclient.login()
         syn.create_snapshot_version(
             dataset.id,
             label="v1.0",
             comment="This is version 1")
     """
     _synapse_entity_type: str = "org.sagebionetworks.repo.model.table.Dataset"
-    _property_keys: List[str] = SchemaBase._property_keys + ['dataset_items']
+    _property_keys: List[str] = SchemaBase._property_keys + ['datasetItems']
+    _local_keys: List[str] = SchemaBase._local_keys + ['folders_to_add', 'force']
 
     def __init__(self, name=None, columns=None, parent=None, properties=None,
-                 annotations=None, local_state=None, dataset_items=None, **kwargs):
-        self.properties.setdefault('dataset_items', [])
+                 annotations=None, local_state=None, dataset_items=None,
+                 folder_ids=None, force=None, **kwargs):
+        self.properties.setdefault('datasetItems', [])
+        self.__dict__.setdefault('folders_to_add', set())
+        self.__dict__.setdefault('force', False)
         super(Dataset, self).__init__(
             name=name, columns=columns, properties=properties,
             annotations=annotations, local_state=local_state, parent=parent,
             **kwargs
         )
+
+        # Add default Dataset columns if schema is not provided.
+        if not self.properties.columnIds and columns is None:
```

Comment:
Are default columns added in `_before_synaspe_store`?

Reply:
Yep!  Its [super()](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/0f0316f57fe2c10e2d8153da67133588e9bb8046/synapseclient/table.py#L712) takes care of this.

Reply:
If it is handled in `super`, why should we add it here?

Reply:
I misunderstood your initial question - I thought you were asking whether these columns are added to the object during `_before_synapse_store`.

But no, none of the other table-like entities set a default schema.  For Dataset, I thought it would be necessary because otherwise, existing items would not show on the UI if a schema is not set.

Reply:
I agree that the default columns should just be added since this is the behavior in the UI. You can use this function to actually get the default columns of a dataset:
```
syn._get_default_view_columns("datasets")
```

Reply:
Oh, wow, didn't know that method existed, thanks!